lua/telescope/utils.lua

```diff
@@ -84,22 +84,20 @@ end
 
 utils.diagnostics_to_tbl = function(opts)
   opts = opts or {}
-
-  local bufnr = vim.api.nvim_get_current_buf()
-  local filename = vim.api.nvim_buf_get_name(bufnr)
-  local buffer_diags = vim.lsp.diagnostic.get(bufnr, opts.client_id)
-
   local items = {}
+  local current_buf = vim.api.nvim_get_current_buf()
   local lsp_type_diagnostic = {[1] = "Error", [2] = "Warning", [3] = "Information", [4] = "Hint"}
-  local insert_diag = function(diag)
+
+  local preprocess_diag = function(diag, bufnr)
+    local filename = vim.api.nvim_buf_get_name(bufnr)
     local start = diag.range['start']
     local finish = diag.range['end']
     local row = start.line
     local col = start.character
 
     local line = (vim.api.nvim_buf_get_lines(bufnr, row, row + 1, false) or {""})[1]
```

Comment:
This line throws a `nil` because `{nil}` is considered as true when I tested it.

```suggestion
    local line = vim.api.nvim_buf_get_lines(bufnr, row, row + 1, false)[1] or ""
```